### PR TITLE
[APPS-2108] date-fns adapter for datetime pickers, many datetime parsing and validation fixes

### PR DESCRIPTION
**Pull request by DenysVuika**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [x] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

- migrate date cloud widget to ADF date-fns provider
- fix unit tests using incorrect date expectations
- **new** ADF DateTime date-fns adapter for mat-datetimepicker component
- migrate DateTime widget to new datetime adapter
- numerous bug fixes to datetime validation and tests
- fixes to storing vs displaying datetime values
- remove useless e2e for demo shell search (testing non-ADF functionality like deleting files from search, viewer, etc)

Screenshot 2023-10-13 at 01 05 19

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
https://alfresco.atlassian.net/browse/APPS-2108
https://alfresco.atlassian.net/browse/APPS-2232

### CODE REVIEW

**Review by swapnil-verma-gl — APPROVED** [review]

File: lib/process-services-cloud/src/lib/form/components/widgets/date/date-cloud.widget.ts [hunk 1]
```diff
@@ -17,22 +17,20 @@
 
 /* eslint-disable @angular-eslint/component-selector */
 
-import { Component, OnInit, ViewEncapsulation, OnDestroy } from '@angular/core';
-import { DateAdapter, MAT_DATE_FORMATS } from '@angular/material/core';
-import moment, { Moment } from 'moment';
+import { Component, OnInit, ViewEncapsulation, OnDestroy, Input } from '@angular/core';
+import { DateAdapter, MAT_DATE_FORMATS/*, MAT_DATE_LOCALE*/ } from '@angular/material/core';
```
Comment (MateuszHY) [inline lib/process-services-cloud/src/lib/form/components/widgets/date/date-cloud.widget.ts:21]: Why is the comment part here?

File: lib/process-services-cloud/src/lib/form/components/widgets/date/date-cloud.widget.ts [hunk 2]
```diff
@@ -50,66 +48,76 @@ import { DATE_FORMAT_CLOUD } from '../../../../models/date-format-cloud.model';
...
-    onDateChanged(newDateValue) {
-        const date = moment(newDateValue, this.field.dateDisplayFormat, true);
-        if (date.isValid()) {
-            this.field.value = date.format(this.field.dateDisplayFormat);
+    onDateChanged(event: MatDatepickerInputEvent<Date>) {
+        const value = event.value;
+        const input = event.targetElement as HTMLInputElement;
+
+        if (value) {
+            // store back to the `dd-MM-yyyy` format
```
Comment (MateuszHY) [inline lib/process-services-cloud/src/lib/form/components/widgets/date/date-cloud.widget.ts:115]: do we need this comment ?

File: lib/process-services-cloud/src/lib/form/components/widgets/date/date-cloud.widget.ts [hunk 3]
```diff
@@ -50,66 +48,76 @@ import { DATE_FORMAT_CLOUD } from '../../../../models/date-format-cloud.model';
 })
 export class DateCloudWidgetComponent extends WidgetComponent implements OnInit, OnDestroy {
     typeId = 'DateCloudWidgetComponent';
+    DATE_FORMAT = 'dd-MM-yyyy';
```
Comment (MichalKinas) [inline lib/process-services-cloud/src/lib/form/components/widgets/date/date-cloud.widget.ts:51]: Maybe we can make it readonly to make sure nobody will modify it in the code?

File: lib/process-services/src/lib/form/widgets/dynamic-table/editors/models/date-cell-validator-model.spec.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        const readonly = { editable: false, type: DateCellValidator.DATE_TYPE } as DynamicTableColumn;
+        expect(validator.isSupported(readonly)).toBeFalse();
+
+        const editable =  { editable: true, type: DateCellValidator.DATE_TYPE } as DynamicTableColumn;
+        expect(validator.isSupported(editable)).toBeTrue();
+    });
+
+    it('should support only date column type', () => {
+        const date = { editable: true, type: DateCellValidator.DATE_TYPE } as DynamicTableColumn;
+        expect(validator.isSupported(date));
```
Comment (MichalKinas) [inline lib/process-services/src/lib/form/widgets/dynamic-table/editors/models/date-cell-validator-model.spec.ts:44]: Missing `toBeTrue()`

File: lib/process-services/src/lib/form/widgets/dynamic-table/editors/models/date-cell-validator-model.ts [hunk 5]
```diff
@@ -17,34 +17,40 @@
...
-            const dateValue = moment(value, 'YYYY-MM-DDTHH:mm:ss.SSSSZ', true);
-            if (!dateValue.isValid()) {
                 if (summary) {
                     summary.isValid = false;
                     summary.message = `Invalid '${column.name}' format.`;
                 }
+
                 return false;
+            } else {
+                return column.required ? false : true;
```
Comment (MichalKinas) [inline lib/process-services/src/lib/form/widgets/dynamic-table/editors/models/date-cell-validator-model.ts:53]: Might just be `!column.required`

File: lib/process-services-cloud/src/lib/form/date-formats.ts
Comment (AleksanderSklorz) [inline lib/process-services-cloud/src/lib/form/date-formats.ts:1]: Just to ensure - do we require this file or can we reuse somehow lib/core/src/lib/form/date-formats.ts?

Comment (DenysVuika) [reply lib/process-services-cloud/src/lib/form/date-formats.ts:1]: the cloud set of formats is different than aps one... for the shared code I've already added another set

**Review by MichalKinas — APPROVED** [review]

**Review by AleksanderSklorz — APPROVED** [review]

**Review by MateuszHY — APPROVED** [review]

**Pull request merged**